### PR TITLE
Update DG

### LINKED ISSUE #103: Update Developer Guide (Mark command)

### PR DESCRIPTION
**Pull request by newway1814**
closes #103 

- [X] Update Developer Guide for Mark Command feature.
- [X] Add UML class diagram for Mark/Unmark Command.
- [X] Add Sequence diagram for Mark command.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ The sequence diagram below illustrates the interactions within the `Logic` compo
 
 How the `Logic` component works:
 
-1. When `Logic` is called upon to execute a command, it is passed to an `AddressBookParser` object which in turn creates a parser that matches the command (e.g., `DeleteCommandParser`) and uses it to parse the command.
+1. When `Logic` is called upon to execute a command, it is passed to an `ProfPlanParser` object which in turn creates a parser that matches the command (e.g., `DeleteCommandParser`) and uses it to parse the command.
```
Comment (eyelessrhyme7) [inline docs/DeveloperGuide.md:104]: Good job in renaming leftover instances of AddressBook to ProfPlan. It is contributions like this that make you a true asset to our team.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -155,6 +155,80 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+### UML Class Diagram
+<div style="display: flex; justify-content: space-between; margin-bottom: 20px;">
+    <div style="flex: 1; margin-right: 5px;">
+        <img src="images/MarkCommandClassDiagram.png" alt="MarkCommand Class Diagram" width="380">
+    </div>
+    <div style="flex: 1;">
+        <img src="images/UnmarkCommandClassDiagram.png" alt="UnmarkCommand Class Diagram" width="380">
+    </div>
+</div>
+
```
Comment (eyelessrhyme7) [inline docs/DeveloperGuide.md:171]: Very nicely structured implementation details for MarkCommand. Easy to understand and follow.

File: docs/diagrams/MarkcommandSequencediagram.puml [hunk 3]
```diff
@@ -0,0 +1,73 @@
+@startuml
```
Comment (eyelessrhyme7) [inline docs/diagrams/MarkcommandSequencediagram.puml:1]: Great job using plantuml to make Sequence diagrams. I have plenty to learn from your habits!

**Review by eyelessrhyme7 — APPROVED** [review]

**Review by jack1e0 — COMMENTED** [review]
LGTM

**Review by yarnmengnus — APPROVED** [review]
LGTM
Nice changes that follow DG guidelines.

**Pull request merged**